### PR TITLE
Stronger integration test for duplication

### PR DESCRIPTION
**Pull request by bharatkashyap**

### CODE REVIEW

File: test/integration/appDuplicate.spec.ts [hunk 1]
```diff
@@ -10,10 +10,18 @@ test('duplicate app from home flow', async ({ page }) => {
   await homeModel.goto();
   const app = await homeModel.createApplication({ dom });
   await homeModel.goto();
-  await homeModel.duplicateApplication(app.name);
+  const duplicateApp = await homeModel.duplicateApplication(app.name);
   await homeModel.goto();
   await expect(page.locator(locators.toolpadHomeAppRow(`${app.name} (copy)`))).toBeVisible();
   await homeModel.duplicateApplication(app.name);
   await homeModel.goto();
```
Comment (apedroferreira) [inline test/integration/appDuplicate.spec.ts:17]: maybe no need to check twice? unless there's something specific about the logic we're trying to cover here

File: test/integration/appDuplicate.spec.ts [hunk 2]
```diff
@@ -10,10 +10,18 @@ test('duplicate app from home flow', async ({ page }) => {
...
   await expect(page.locator(locators.toolpadHomeAppRow(`${app.name} (copy 2)`))).toBeVisible();
+
+  await page.goto(`/_toolpad/app/${duplicateApp.id}/preview/pages`);
```
Comment (apedroferreira) [inline test/integration/appDuplicate.spec.ts:20]: the `ToolpadRuntime` model might be useful for this part
checkout out the `components` integration test, maybe this test could do something similar (if it's not working)

File: test/integration/appDuplicate.spec.ts (hunk 1 above)
Comment (bharatkashyap) [reply test/integration/appDuplicate.spec.ts:17]: Trying to check if the second duplicate is appropriately named

File: test/integration/appDuplicate.spec.ts (hunk 2 above)
Comment (bharatkashyap) [reply test/integration/appDuplicate.spec.ts:20]: Thanks! I'll check that out

File: test/integration/appDuplicate.spec.ts (hunk 1 above)
Comment (apedroferreira) [reply test/integration/appDuplicate.spec.ts:17]: got it, makes sense then!

**Pull request closed without merging**